include/seqan3/io/sam_file/detail/format_sam_base.hpp

```diff
@@ -461,51 +445,125 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
             } break;
 
             case 'S': /* SQ (sequence dictionary) tag */
             {
+                skip_until_predicate(is_char<'\t'>);                    // skip SQ
+
                 ref_info_present_in_header = true;
                 std::ranges::range_value_t<decltype(hdr.ref_ids())> id;
                 std::tuple<int32_t, std::string> info{};
 
-                parse_tag_value(id);                                    // parse required SN (sequence name) tag
-                ++it;                                                   // skip tab or newline
-                parse_tag_value(get<0>(info));                          // parse required LN (length) tag
-
-                if (is_char<'\t'>(*it))                                 // read rest of the tags
+                // All tags can appear in any order, SN and LN are required tags
+                while (is_char<'\t'>(*it))
                 {
                     ++it;                                               // skip tab
-                    parse_until_newline(get<1>(info));
+
+                    switch (*it)
+                    {
+                        case 'S':
+                        {
+                            ++it;                                       // skip S
+                            if (is_char<'N'>(*it))                      // parse required SN (sequence name) tag
+                            {
+                                parse_tag_value(id);
+                            }
+                            else if (is_char<'P'>(*it))                 // SP (Species) tag
+                            {
+                                parse_until_tab_or_newline(get<1>(info));
+                            }
+                            else
+                                throw format_error{std::string{"Illegal SAM header tag: V"} + *it};
+                        } break;
+                        case 'L':
+                        {
+                            ++it;                                       // skip L
+                            if (is_char<'N'>(*it))                      // parse required LN (length) tag
+                                parse_tag_value(get<0>(info));
+                            else
+                                throw format_error{std::string{"Illegal SAM header tag: S"} + *it};
+                        } break;
+                        case 'A':                                       // read rest of the tags (AH, AS
+                        case 'D':                                       // DS,
+                        case 'M':                                       // M5,
+                        case 'T':                                       // TP,
+                        case 'U':                                       // UR)
+                        {
+                            parse_until_tab_or_newline(get<1>(info));
```

Comment:
This will only store the last tag if there are multiple ones because you always parse until tab and overwrite `get<1>(info)`, no?

Reply:
No it apends the tags. Thats why I need to `push_back('\t’)`